packages/shared/src/lib/auxiliary/wallet-connect/handlers/onSessionRequest.handler.ts

```diff
@@ -58,23 +59,28 @@ export function onSessionRequest(event: Web3WalletTypes.SessionRequest): void {
         return
     }
 
-    const signAndSend = method === 'eth_sendTransaction'
-
     switch (method) {
-        case 'eth_sendTransaction':
-        case 'eth_signTransaction':
-            void handleEthTransaction(request.params[0], dapp, chain, signAndSend, returnResponse, verifiedState)
+        case RpcMethod.EthSendTransaction:
+        case RpcMethod.EthSignTransaction:
+        case RpcMethod.EthSendRawTransaction: {
+            const evmTransactionData =
+                method === RpcMethod.EthSendRawTransaction
+                    ? getEvmTransactionFromHexString(request.params[0])
```

Comment:
Maybe the if statement can also move in the helper function? If you do the function, this should be renamed

Reply:
I feel like this function should be agnostic to the context, and just transform a transaction string to an object